clang/lib/CIR/Dialect/Transforms/LoweringPrepare.cpp

```diff
@@ -1120,6 +1125,78 @@ void LoweringPreparePass::lowerIterEndOp(IterEndOp op) {
   op.erase();
 }
 
+void LoweringPreparePass::lowerStdInitializerListOp(StdInitializerListOp op) {
+  auto loc = op.getLoc();
+  cir::CIRDataLayout dataLayout(theModule);
+  auto ptrDiffTy = cir::IntType::get(&getContext(), 64, /*isSigned=*/false);
+
+  auto args = op.getArgs();
+
+  auto stdInitializerListType =
+      mlir::cast<cir::StructType>(op.getInitList().getType().getPointee());
+  clang::RecordDecl::field_range stdInitializerListFields =
+      stdInitializerListType.getAst().getRawDecl()->fields();
+  assert(llvm::range_size(stdInitializerListFields) == 2U);
+
+  mlir::Type elementType =
+      mlir::cast<cir::PointerType>(stdInitializerListType.getMembers()[0])
+          .getPointee();
+  auto tempArrayType =
+      cir::ArrayType::get(&getContext(), elementType, args.size());
+  auto alignment = mlir::IntegerAttr::get(
+      mlir::IntegerType::get(&getContext(), 64),
+      dataLayout.getPrefTypeAlign(tempArrayType).value());
+
+  CIRBaseBuilderTy builder(getContext());
+  builder.setInsertionPointAfter(op);
+
+  mlir::Value arrayPtr = builder.createAlloca(
+      loc, cir::PointerType::get(tempArrayType), tempArrayType, "", alignment);
```

Comment:
Add a mark `assert(!cir::MissingFeatures::addressSpace());` here if you don't want to bother address space.